homeassistant/components/climacell/weather.py

```diff
@@ -0,0 +1,305 @@
+    CC_ATTR_TEMPERATURE_HIGH,
+    CC_ATTR_TEMPERATURE_LOW,
+    CC_ATTR_TIMESTAMP,
+    CC_ATTR_VISIBILITY,
+    CC_ATTR_WIND_DIRECTION,
+    CC_ATTR_WIND_SPEED,
+    CLEAR_CONDITIONS,
+    CONDITIONS,
+    CONF_FORECAST_TYPES,
+    CONF_TIMESTEP,
+    CURRENT,
+    DAILY,
+    DOMAIN,
+    FORECASTS,
+    HOURLY,
+    NOWCAST,
+)
+
+# mypy: allow-untyped-defs, no-check-untyped-defs
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _translate_condition(
+    condition: Optional[str], sun_is_up: bool = True
+) -> Optional[str]:
+    """Translate ClimaCell condition into an HA condition."""
+    if not condition:
+        return None
+    if "clear" in condition.lower():
+        if sun_is_up:
+            return CLEAR_CONDITIONS["day"]
+        return CLEAR_CONDITIONS["night"]
+    return CONDITIONS[condition]
+
+
+def _forecast_dict(
+    hass: HomeAssistantType,
+    time: str,
+    use_datetime: bool,
+    condition: str,
+    precipitation: Optional[float],
+    precipitation_probability: Optional[float],
+    temp: Optional[float],
+    temp_low: Optional[float],
+    wind_direction: Optional[float],
+    wind_speed: Optional[float],
+) -> Dict[str, Any]:
+    """Return formatted Forecast dict from ClimaCell forecast data."""
+    if use_datetime:
+        translated_condition = _translate_condition(
+            condition,
+            is_up(hass, dt_util.parse_datetime(time).replace(tzinfo=pytz.utc)),
+        )
+    else:
+        translated_condition = _translate_condition(condition, True)
+
+    if hass.config.units.is_metric:
+        if precipitation:
+            precipitation = (
+                distance_convert(precipitation / 12, LENGTH_FEET, LENGTH_METERS) * 1000
+            )
+        if temp:
+            temp = temp_convert(temp, TEMP_FAHRENHEIT, TEMP_CELSIUS)
+        if temp_low:
+            temp_low = temp_convert(temp_low, TEMP_FAHRENHEIT, TEMP_CELSIUS)
+        if wind_speed:
+            wind_speed = distance_convert(wind_speed, LENGTH_MILES, LENGTH_KILOMETERS)
+
+    data = {
+        ATTR_FORECAST_TIME: time,
+        ATTR_FORECAST_CONDITION: translated_condition,
+        ATTR_FORECAST_PRECIPITATION: precipitation,
+        ATTR_FORECAST_PRECIPITATION_PROBABILITY: precipitation_probability,
+        ATTR_FORECAST_TEMP: temp,
+        ATTR_FORECAST_TEMP_LOW: temp_low,
+        ATTR_FORECAST_WIND_BEARING: wind_direction,
+        ATTR_FORECAST_WIND_SPEED: wind_speed,
+    }
+
+    return {k: v for k, v in data.items() if v is not None}
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    config_entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up a config entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = [
+        ClimaCellWeatherEntity(config_entry, coordinator, forecast_type)
+        for forecast_type in [DAILY, HOURLY, NOWCAST]
+    ]
+    async_add_entities(entities, update_before_add=True)
```

Comment:
The entity doesn't have an `async_update` method.
```suggestion
    async_add_entities(entities)
```